client/my-sites/checkout/controller.jsx

```diff
@@ -288,19 +286,16 @@ export function upsellNudge( context, next ) {
 }
 
 export function upsellRedirect( context, next ) {
-	const { receiptId, site, upsellMeta, upsellType } = context.params;
+	const { receiptId, site /*, upsellMeta, upsellType */ } = context.params;
 
 	setSectionMiddleware( { name: 'checkout-offer-redirect' } )( context );
 
 	let upsellExperimentName;
 	let upsellExperimentAssignmentName;
 	let upsellUrl;
 
-	if ( PROFESSIONAL_EMAIL_OFFER === upsellType ) {
-		upsellExperimentName = 'calypso_promote_professional_email_post_checkout_2022_02';
-		upsellExperimentAssignmentName = 'treatment';
-		upsellUrl = `/checkout/offer-professional-email/${ upsellMeta }/${ receiptId }/${ site }`;
-	}
+	// When next we need a redirect based on A/B test, add any logic based on upsellType here
+	// While this code block is empty, this function is effectively a no-op.
```

Comment:
Thank's for this! ❤️

Reply:
I wonder if we shouldn't actually keep the code but just comment it out:

```javascript
// When next we need a redirect based on A/B test, add any logic based on upsellType here, e.g.:
//
// if ( PROFESSIONAL_EMAIL_OFFER === upsellType ) {
//   upsellExperimentName = 'calypso_promote_professional_email_post_checkout_2022_02';
//   upsellExperimentAssignmentName = 'treatment';
//   upsellUrl = `/checkout/offer-professional-email/${ upsellMeta }/${ receiptId }/${ site }`;
// }
```

That way it will be more obvious for other developers to know how to use this middleware.